### PR TITLE
feat: add note and edit note functionality added

### PR DESCRIPTION
**Pull request by devdeadviz**
# Change Logs
- Add new note functionality added.
- Edit existing note data functionality added.
## Deploy Preview link: https://deploy-preview-4--devdeadviz-my-notes.netlify.app/

### CODE REVIEW

File: src/components/NoteCard/NoteCard.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import "./NoteCard.css";
 
-const NoteCard = ({ title, body }) => {
+const NoteCard = ({ note, editNoteHandler }) => {
+  const { title, body, createdAt } = note;
```
Comment (ivinayakg) [inline src/components/NoteCard/NoteCard.js:4]: nice that you separated the destructuring of this model

File: src/components/NoteForm/NoteForm.js [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
+  } = useNote();
+
   return (
     <>
-      <form className="note-form-wrapper my-4">
+      <form
+        className="note-form-wrapper my-4"
+        onSubmit={
+          editedNotes
+            ? (e) => updateNoteHandler(e, { ...note, createdAt: getFormattedDate() })
```
Comment (ivinayakg) [inline src/components/NoteForm/NoteForm.js:16]: if yu can then put this function in a separate function not as a dynamic function in the jsx

**Review by ivinayakg — APPROVED** [review]
good clean, code, good job! 👍

File: src/components/NoteCard/NoteCard.js (hunk 1 above)
Comment (devdeadviz) [reply src/components/NoteCard/NoteCard.js:4]: Thank you!

File: src/components/NoteForm/NoteForm.js (hunk 2 above)
Comment (devdeadviz) [reply src/components/NoteForm/NoteForm.js:16]: I will look into it, thanks for pointing.

**Review by xii-vi — APPROVED** [review]
_Clean code as well as the UI.
like the way you did proper restructring_

**Pull request merged**